### PR TITLE
[21383] Fix tests on Ubuntu 24.04

### PR DESCRIPTION
**Pull request by richiware**
## Description
Running this repository's tests fail on Ubuntu 24.04 due to SWIG 4.1. This PR fixes them.

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally
- [x] Changes are API compatible. 
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: fastdds_python/test/api/test_datawriter.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ def test_get_set_listener(datawriter):
     assert(datawriter.get_listener() == listener)
     assert(fastdds.StatusMask.all() == datawriter.get_status_mask())
 
-    def test(status_mask_1, status_mask_2):
+    def test(status_mask_1):
         """
         Test the entity creation using the two types of StatusMasks.
```
Comment (JesusPoderoso) [inline fastdds_python/test/api/test_datawriter.py:258]:
```suggestion
        Test the entity creation using the type of StatusMask.
```

File: fastdds_python/test/api/test_datawriter.py [hunk 2]
```diff
@@ -263,57 +263,24 @@ def test(status_mask_1, status_mask_2):
...
-        fastdds.StatusMask_offered_deadline_missed() << \
-        fastdds.StatusMask_offered_incompatible_qos() << \
-        fastdds.StatusMask_publication_matched() << \
-        fastdds.StatusMask_requested_deadline_missed() << \
-        fastdds.StatusMask_requested_incompatible_qos() << \
-        fastdds.StatusMask_sample_lost() << \
-        fastdds.StatusMask_sample_rejected() << \
-        fastdds.StatusMask_subscription_matched()
+    test(fastdds.StatusMask.all())
+    test(fastdds.StatusMask.all())
```
Comment (JesusPoderoso) [inline fastdds_python/test/api/test_datawriter.py:269]: Isn't it duplicated?
```suggestion
```

File: fastdds_python/test/api/test_datareader.py [hunk 3]
```diff
@@ -743,67 +743,34 @@ def test_get_set_listener(datareader):
...
-        fastdds.StatusMask_offered_deadline_missed() << \
-        fastdds.StatusMask_offered_incompatible_qos() << \
-        fastdds.StatusMask_publication_matched() << \
-        fastdds.StatusMask_requested_deadline_missed() << \
-        fastdds.StatusMask_requested_incompatible_qos() << \
-        fastdds.StatusMask_sample_lost() << \
-        fastdds.StatusMask_sample_rejected() << \
-        fastdds.StatusMask_subscription_matched()
+    test(fastdds.StatusMask.all())
+    test(fastdds.StatusMask.all())
```
Comment (JesusPoderoso) [inline fastdds_python/test/api/test_datareader.py:759]: Isn't it duplicated?
```suggestion
```

File: fastdds_python/test/api/test_subscriber.py [hunk 4]
```diff
@@ -450,67 +377,34 @@ def test_get_set_listener(subscriber):
...
-        fastdds.StatusMask_offered_deadline_missed() << \
-        fastdds.StatusMask_offered_incompatible_qos() << \
-        fastdds.StatusMask_publication_matched() << \
-        fastdds.StatusMask_requested_deadline_missed() << \
-        fastdds.StatusMask_requested_incompatible_qos() << \
-        fastdds.StatusMask_sample_lost() << \
-        fastdds.StatusMask_sample_rejected() << \
-        fastdds.StatusMask_subscription_matched()
+    test(fastdds.StatusMask.all())
+    test(fastdds.StatusMask.all())
```
Comment (JesusPoderoso) [inline fastdds_python/test/api/test_subscriber.py:393]: Isn't it duplicated?
```suggestion
```

File: fastdds_python/test/api/test_datareader.py [hunk 5]
```diff
@@ -743,67 +743,34 @@ def test_get_set_listener(datareader):
     assert(fastdds.StatusMask.all() ==
            datareader.get_status_mask())
 
-    def test(status_mask_1, status_mask_2):
+    def test(status_mask_1):
```
Comment (JesusPoderoso) [inline fastdds_python/test/api/test_datareader.py:746]: I think that there is no need to keep the `_1` suffix anymore.
If the suggestion is accepted, please apply it to all changes/files.
```suggestion
    def test(status_mask):
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
A couple of nitpicks

File: fastdds_python/test/api/test_datawriter.py (hunk 1 above)
Comment (richiware) [reply fastdds_python/test/api/test_datawriter.py]: Done in ddb9b93

File: fastdds_python/test/api/test_datawriter.py (hunk 2 above)
Comment (richiware) [reply fastdds_python/test/api/test_datawriter.py]: Done in ddb9b93

File: fastdds_python/test/api/test_datareader.py (hunk 3 above)
Comment (richiware) [reply fastdds_python/test/api/test_datareader.py]: Done in ddb9b93

File: fastdds_python/test/api/test_subscriber.py (hunk 4 above)
Comment (richiware) [reply fastdds_python/test/api/test_subscriber.py]: Done in ddb9b93

File: fastdds_python/test/api/test_datareader.py (hunk 5 above)
Comment (richiware) [reply fastdds_python/test/api/test_datareader.py]: Done in ddb9b93

**Review by JesusPoderoso — APPROVED** [review]
LGTM  with green CI

**Pull request merged**